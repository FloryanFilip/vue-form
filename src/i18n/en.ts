export default {
	common: {
		navigateToForm: 'Navigate to form',
		confirm: 'Confirm',
		cancel: 'Cancel',
		success: 'Success',
	},
	buttons: {
		defaultLabel: 'Default label',
		confirm: '@:common.confirm',
		cancel: '@:common.cancel',
	},
	createNewPersonForm: {
		title: 'Add new person',
		labels: {
			fullName: 'First and last name',
			email: 'Email address',
			phone: 'Contact number',
		},
		placeholders: {
			fullName: 'John Wick',
			email: 'John.wick@excommunicado.com',
			phone: '+100 1-1-1',
		},
	},
	validations: {
		minLength: 'Minimal length is: {count}',
		maxLength: 'Maximal length is: {count}',
		required: '{value} is required | {value} is required',
		invalidPattern: 'Wrong format',
	},
};
