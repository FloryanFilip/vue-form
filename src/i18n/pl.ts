export default {
	common: {
		navigateToForm: 'Przejdź do formularza',
		confirm: 'Potwierdź',
		cancel: 'Anuluj',
		success: 'Sukces',
	},
	buttons: {
		defaultLabel: 'Domyślna etykieta',
		confirm: '@:common.confirm',
		cancel: '@:common.cancel',
	},
	createNewPersonForm: {
		title: 'Dodaj nową osobę',
		labels: {
			fullName: 'Imię i nazwisko',
			email: 'Adres Email',
			phone: 'Telefon kontaktowy',
		},
		placeholders: {
			fullName: 'Jan Kowalski',
			email: 'jan.kowalski@gmail.com',
			phone: '+48 533-533-533',
		},
	},
	validations: {
		minLength: 'Minimalna dopuszczalna długość wynosi: {count}',
		maxLength: 'Maksymalna dopuszczalna długość wynosi: {count}',
		required: '{value} jest wymagane | {value} jest wymagany',
		invalidPattern: 'Zły format',
	},
};
