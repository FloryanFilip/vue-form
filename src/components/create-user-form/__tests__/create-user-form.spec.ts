import CreateUserFormComponent from '@/components/create-user-form/create-user-form.component.vue';
import { createLocalVue, shallowMount, Wrapper } from '@vue/test-utils';
import { useI18n } from '../../../../tests/unit/__utils__/i18n';

describe('CreateUserFormComponent', () => {
	let wrapper!: Wrapper<CreateUserFormComponent>;
	const localVue = createLocalVue();
	const i18n = useI18n(localVue);

	const factory = (props?: object) => {
		return shallowMount(CreateUserFormComponent, {
			localVue,
			i18n,
			propsData: {
				fullName: {
					value: '',
					error: undefined,
				},
				email: {
					value: '',
					error: undefined,
				},
				phoneNumber: {
					value: '',
					error: undefined,
				},
				loading: false,
				...props,
			},
			stubs: {
				'app-input': true,
				'app-button': true,
			},
		});
	};

	beforeEach(() => wrapper = factory());
	afterEach(() => wrapper.destroy());

	it('should match snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should display fullName label', () => {
		const fullNameLabel = wrapper.find('[data-testid="fullName"]');

		expect(fullNameLabel.text()).toBe('ImiÄ™ i nazwisko');
	});

	it('should display email label', () => {
		const fullNameLabel = wrapper.find('[data-testid="email"]');

		expect(fullNameLabel.text()).toBe('Adres Email');
	});

	it('should display phoneNumber label', () => {
		const fullNameLabel = wrapper.find('[data-testid="phoneNumber"]');

		expect(fullNameLabel.text()).toBe('Telefon kontaktowy');
	});

	it('should emit [submitForm] event on form submit', async () => {
		await wrapper.find('form').trigger('submit');

		expect(wrapper.emitted().submitForm).toBeTruthy();
		expect(wrapper.emitted().submitForm).toHaveLength(1);
		expect(wrapper.emitted().submitForm![0]).toEqual([]);
	});

	it('should emit [cancel] event on cancel button click', () => {
		wrapper.find('[data-testid="button-cancel"]').vm.$emit('click');

		expect(wrapper.emitted().cancel).toBeTruthy();
		expect(wrapper.emitted().cancel).toHaveLength(1);
		expect(wrapper.emitted().cancel![0]).toEqual([]);
	});

	it('should emit [setFullName] event on fullName input change', () => {
		wrapper.find('[data-testid="fullName"]').vm.$emit('input', { target: { value: 'Jan'} });

		expect(wrapper.emitted().setFullName).toBeTruthy();
		expect(wrapper.emitted().setFullName).toHaveLength(1);
		expect(wrapper.emitted().setFullName![0]).toEqual(['Jan']);
	});

	it('should emit [setEmail] event on email input change', () => {
		wrapper.find('[data-testid="email"]').vm.$emit('input', { target: { value: 'kowalski@wp.pl'} });

		expect(wrapper.emitted().setEmail).toBeTruthy();
		expect(wrapper.emitted().setEmail).toHaveLength(1);
		expect(wrapper.emitted().setEmail![0]).toEqual(['kowalski@wp.pl']);
	});

	it('should emit [setPhoneNumber] event on phoneNumber input change', () => {
		wrapper.find('[data-testid="phoneNumber"]').vm.$emit('input', { target: { value: '+48533533533'} });

		expect(wrapper.emitted().setPhoneNumber).toBeTruthy();
		expect(wrapper.emitted().setPhoneNumber).toHaveLength(1);
		expect(wrapper.emitted().setPhoneNumber![0]).toEqual(['+48533533533']);
	});
});
