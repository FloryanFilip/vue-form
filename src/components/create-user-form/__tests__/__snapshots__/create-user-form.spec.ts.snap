// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CreateUserFormComponent should match snapshot 1`] = `
<div class="container">
  <form class="form">
    <div class="row">
      <div class="title">
        Dodaj nową osobę
      </div>
    </div>
    <div class="row">
      <app-input-stub type="text" placeholder="Jan Kowalski" value="" data-testid="fullName" class="">
        Imię i nazwisko
      </app-input-stub>
    </div>
    <div class="row">
      <app-input-stub type="text" placeholder="jan.kowalski@gmail.com" value="" data-testid="email" class="">
        Adres Email
      </app-input-stub>
    </div>
    <div class="row">
      <app-input-stub type="text" placeholder="+48 533-533-533" value="" data-testid="phoneNumber" class="">
        Telefon kontaktowy
      </app-input-stub>
    </div>
    <div class="row">
      <app-button-stub type="submit" data-testid="button-submit" class="button-primary button--full-width"><span>Potwierdź</span></app-button-stub>
      <app-button-stub type="button" data-testid="button-cancel" class="button-secondary button--full-width">Anuluj
      </app-button-stub>
    </div>
  </form>
</div>
`;
